pkg/backuphelpers/backupvars.go

```diff
@@ -0,0 +1,92 @@
+package backuphelpers
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
+	backupv1alpha1 "github.com/openshift/api/config/v1alpha1"
+	prune "github.com/openshift/cluster-etcd-operator/pkg/cmd/prune-backups"
+)
+
+type Enqueueable interface {
+	Enqueue()
+}
+
+type BackupVar interface {
+	AddListener(listener Enqueueable)
+	SetBackupSpec(spec *backupv1alpha1.EtcdBackupSpec)
+	ArgString() string
+}
+
+type BackupConfig struct {
+	enabled   bool
+	spec      *backupv1alpha1.EtcdBackupSpec
+	listeners []Enqueueable
+	mux       sync.Mutex
+}
+
+func NewDisabledBackupConfig() *BackupConfig {
+	return &BackupConfig{
+		enabled: false,
+		mux:     sync.Mutex{},
+	}
+}
+
+func (b *BackupConfig) SetBackupSpec(spec *backupv1alpha1.EtcdBackupSpec) {
+	b.mux.Lock()
+	defer b.mux.Unlock()
+
+	if reflect.DeepEqual(b.spec, spec) {
+		return
+	}
+
+	b.enabled = spec != nil
+
+	b.spec = spec
+	for _, l := range b.listeners {
+		l.Enqueue()
+	}
+}
+
+func (b *BackupConfig) ArgString() string {
+	b.mux.Lock()
+	defer b.mux.Unlock()
+
+	args := []string{"    args:"}
+	args = append(args, fmt.Sprintf("- --%s=%v", "enabled", b.enabled))
+
+	if b.spec == nil {
+		return strings.Join(args, "\n    ")
+	}
```

Comment:
```suggestion
	args = append(args, fmt.Sprintf("- --%s=%v", "enabled", b.enabled))

	if !b.enabled {
		return strings.Join(args, "\n    ")
	}
```

if you already create an extra variable, make use of it :)

Reply:
+1